### ISSUE TITLE
Figure out why Node gyp fails at building some packages

### ISSUE THREAD

**Issue by Jarmahent** [Jarmahent/Clippy#9]
Stack Trace:

```
Error: C:\Program Files\nodejs\node.exe exited with code 1
Output:
yarn install v1.9.4
[1/4] Resolving packages...
success Already up-to-date.
$ yarn electron-rebuild
yarn run v1.9.2
$ node -r babel-register ../internals/scripts/ElectronRebuild.js
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

Error output:

An unhandled error occurred inside electron-rebuild
gyp info it worked if it ends with ok
gyp info using node-gyp@3.7.0
gyp info using node@10.9.0 | win32 | x64
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Python27\python.exe -c import platform; print(platform.python_version());
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:289:12)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at maybeClose (internal/child_process.js:961:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:250:5)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\mydata\\clippy\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=2.0.6" "--arch=x64" "--dist-url=h
ttps://atom.io/download/electron" "--build-from-source"
gyp ERR! cwd D:\mydata\clippy\app\node_modules\better-sqlite3
gyp ERR! node -v v10.9.0
gyp ERR! node-gyp -v v3.7.0
gyp ERR! not ok

Failed with exit code: 1

Error: gyp info it worked if it ends with ok
gyp info using node-gyp@3.7.0
gyp info using node@10.9.0 | win32 | x64
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Python27\python.exe -c import platform; print(platform.python_version());
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:289:12)
gyp ERR! stack     at ChildProcess.emit (events.js:182:13)
gyp ERR! stack     at maybeClose (internal/child_process.js:961:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:250:5)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\mydata\\clippy\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--target=2.0.6" "--arch=x64" "--dist-url=h
ttps://atom.io/download/electron" "--build-from-source"
gyp ERR! cwd D:\mydata\clippy\app\node_modules\better-sqlite3
gyp ERR! node -v v10.9.0
gyp ERR! node-gyp -v v3.7.0
gyp ERR! not ok

Failed with exit code: 1
    at SafeSubscriber._error (D:\mydata\clippy\node_modules\spawn-rx\lib\src\index.js:277:84)
    at SafeSubscriber.__tryOrUnsub (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:242:16)
    at SafeSubscriber.error (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:201:26)
    at Subscriber._error (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:132:26)
    at Subscriber.error (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:106:18)
    at MapSubscriber.Subscriber._error (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:132:26)
    at MapSubscriber.Subscriber.error (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:106:18)
    at SafeSubscriber._next (D:\mydata\clippy\node_modules\spawn-rx\lib\src\index.js:251:65)
    at SafeSubscriber.__tryOrSetError (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:251:16)
    at SafeSubscriber.next (D:\mydata\clippy\node_modules\rxjs\Subscriber.js:191:27)
child_process.js:640
    throw err;
    ^
```

I've tried following [better-sqlite3 troubleshooting guide](https://github.com/JoshuaWise/better-sqlite3/blob/master/docs/troubleshooting.md)

Still nothing

**Comment by Jarmahent** [comment]
I have tried so far:

`npm install --global --production windows-build-tools`

Does not work

--------------------------------------------------------------

` npm config set msvs_version 2017`

Does not work

--------------------------------------------------------------


`Installing Microsoft visual build tools`

Does not work



--------------------------------------------------------------


Wiping my entire computer

Does not work

--------------------------------------------------------------

**Comment by Jarmahent** [comment]
Finally fixed the issue:

1. Delete node modules folder

2. Open PowerShell with admin privileges and run:

 
```
npm install --global --production --vs2015 --add-python-to-path windows-build-tools
npm install --global --production --add-python-to-path windows-build-tools node-gyp
```

3.  Go to `~/.npmrc` and make sure these are the settings:

```
msvs_version=2015
python=python2.7
```

**Comment by mishakrul** [comment]
> Finally fixed the issue:
> ```
> npm install --global --production --vs2015 --add-python-to-path windows-build-tools
> ```

Unfortunately this command will just hang if you have Visual Studio 2015 Community Edition installed.  See issue [#47](https://github.com/felixrieseberg/windows-build-tools/issues/47#issuecomment-362157543) for windows-build-tools.  Powershell will freeze up at **"Still waiting for installer log files..."** until the process is killed.  I checked the build logs and it shows the exact same message with regards to the 2015 edition:

![stopblock](https://user-images.githubusercontent.com/16310249/51250278-e4a53600-1963-11e9-95de-3eecbddac061.JPG)

Once I uninstalled 2015 and ran the following command, I began to see the `build-tools-log`s populating in the `.windows-build-tools` directory:

`npm install --global --production windows-build-tools@1.3.2`
